modules/telemetry/telemetry/src/log.cpp

```diff
@@ -28,36 +30,49 @@ class Logger final {
   auto operator=(const Logger&) -> Logger& = delete;
   auto operator=(Logger&&) -> Logger& = delete;
 
-  static void registerSink(std::unique_ptr<heph::telemetry::ILogSink> sink);
+  static void registerSink(std::unique_ptr<ILogSink> sink);
 
-  static void log(heph::telemetry::LogEntry&& log_entry);
+  static void log(LogEntry&& log_entry);
 
 private:
   [[nodiscard]] static auto instance() -> Logger&;
 
   /// @brief Do the actual logging. Take in LogEntry and send it to all sinks
   /// @param LogEntry, class that takes in the structured logs and formats them.
   /// @return void
-  void processLogEntries(const heph::telemetry::LogEntry& entry);
+  void processEntry(const LogEntry& entry);
+
+  /// @brief Empty the queue so that remaining messages get processed
+  void emptyQueue();
 
 private:
   absl::Mutex sink_mutex_;
   // This is used as registry for the log sinks
-  std::vector<std::unique_ptr<heph::telemetry::ILogSink>> sinks_ ABSL_GUARDED_BY(sink_mutex_);
-  heph::concurrency::MessageQueueConsumer<heph::telemetry::LogEntry> log_entries_consumer_;
+  std::vector<std::unique_ptr<ILogSink>> sinks_ ABSL_GUARDED_BY(sink_mutex_);
+  heph::containers::BlockingQueue<LogEntry> entries_;
+  std::future<void> message_process_future_;
 };
 
-Logger::Logger()
-  : log_entries_consumer_([this](const heph::telemetry::LogEntry& entry) { processLogEntries(entry); },
-                          std::nullopt) {
-  log_entries_consumer_.start();
+Logger::Logger() : entries_{ std::nullopt } {
+  message_process_future_ = std::async(std::launch::async, [this]() {
+    while (true) {
+      auto message = entries_.waitAndPop();
+      if (!message.has_value()) {
+        emptyQueue();
+        return;
+      }
+
+      processEntry(message.value());
+    }
+  });
 }
 
 Logger::~Logger() {
   try {
-    log_entries_consumer_.stop().get();
+    entries_.stop();
+    message_process_future_.get();
   } catch (const std::exception& ex) {
-    std::cerr << "While emptying message consumer, exception happened: " << ex.what() << "\n";
+    std::cerr << "While emptying log queue, exception happened: " << ex.what() << "\n";
```

Comment:
given that we are updating this, for consistency I think it is better to:
```suggestion
    fmt::println(stderr, "While emptying log queue, exception happened: {}", ex.what());
```